### PR TITLE
Refactoring Pull Request

### PR DESCRIPTION
**Pull request by AKannanMSFT**
I will be following [architecture overview](https://github.com/microsoft/finnts/wiki/FinnTS-Revised-Architecture) for reference.
This PR is ready for review. We are at the first pass of code-checkin.

### CODE REVIEW

File: R/configure_forecast_run.R [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+#' Gets the back testing spacing
+#' 
+#' Checks if back_test_spacing is set to auto and gets the right one
+#' 
+#' 
+#' @param back_test_spacing back_test_spacing override
+#' @param date_type year, quarter, month, week, day
+#' 
+#' @return Returns back_test_spacing
+get_back_test_spacing <- function(back_test_spacing,
```
Comment (mitokic) [inline R/configure_forecast_run.R:169]: should this be a switch statement instead of multiple if statements?

File: R/forecast_models.R [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+  if(combo_value != "All-Data") {
+    
+    full_data_tbl %>%
+      dplyr::filter(Combo == combo_value)
+    
+  } else {
+    
+    full_data_tbl %>%
+      tidyr::separate(col = 'Combo', 
+                      into = combo_variables,
```
Comment (mitokic) [inline R/forecast_models.R:21]: I'm getting an error when running your latest updates. Seems the error how to do with "combo_variables" not being a recognized object in this function. Looks like combo_variables should be added to function arguments. 

@AKannanMSFT

File: R/forecast_models.R [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+                                      xregs_future_values_list,
+                                      fourier_periods,
+                                      lag_periods,
+                                      rolling_window_periods,
+                                      hist_end_date,
+                                      date_type,
+                                      forecast_horizon,
+                                      run_model_parallel,
+                                      parallel_processing,
+                                      parallel_init_func,
```
Comment (mitokic) [inline R/forecast_models.R:232]: should parallel_init_func and parallel_exit_func be changed to "init_azure_batch_parallel_within", and "exit_azure_batch_parallel_within"

On second thought it looks like you are working on creating those parallel functions. 

@AKannanMSFT

File: R/configure_forecast_run.R (hunk 1 above)
Comment (AKannanMSFT) [reply R/configure_forecast_run.R:169]: Yes. Fixed! I will push in next round.

File: R/forecast_models.R (hunk 2 above)
Comment (AKannanMSFT) [reply R/forecast_models.R:21]: Great catch. Fixed will push in next round

**Pull request merged**